### PR TITLE
Modified ALLLIBDIRS to just have paths with no file names. Used for l…

### PR DESCRIPTION
**Pull request by scsides**
Modified so the loader does not throw warnings when it sees a file name at the end of a -L lib_search_path.

## Related Issue

#3886

## Motivation and Context

Properly populates the loader -L and -l options, so warnings are not thrown

## How Has This Been Tested?

A complete build was done on local and dev systems.

## Types of changes

- [X] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Documentation change (update to the documentation; no code change)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have read and agree to abide by the [Code of Conduct](https://usgs-astrogeology.github.io/code/)
- [X] I have read the [**CONTRIBUTING**](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CONTRIBUTING.md) document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.
- [ ] I have added myself to the [.zenodo.json](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/.zenodo.json) document.
- [ ] I have added any user impacting changes to the [CHANGELOG.md](https://github.com/USGS-Astrogeology/ISIS3/blob/dev/CHANGELOG.md) document.

## Licensing
This project is mostly composed of free and unencumbered software released into the public domain, and we are unlikely to accept contributions that are not also released into the public domain. Somewhere near the top of each file should have these words:

> This work is free and unencumbered software released into the public domain. In jurisdictions that recognize copyright laws, the author or authors of this software dedicate any and all copyright interest in the software to the public domain.

- [X] I dedicate any and all copyright interest in this software to the public domain. I make this dedication for the benefit of the public at large and to the detriment of my heirs and successors. I intend this dedication to be an overt act of relinquishment in perpetuity of all present and future rights to this software under copyright law.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -30,6 +30,10 @@ update the Unreleased link so that it compares against the latest release tag.
 
 ## [Unreleased]
 
+### Fixed
+
+ - Removed the library file names from the full library path and file name before it was added to ALLLIBDIRS for the loader -L flag. [#3886](https://github.com/USGS-Astrogeology/ISIS3/issues/3986)
+
```
Comment (jessemapel) [inline CHANGELOG.md]: This isn't a change that will impact users, I don't think it should be in the changelog.

**Review by jessemapel — COMMENTED** [review]
Small note about the changelog, but this looks good. I'm looking forward to not clearing my console after every successful build.

File: CHANGELOG.md (hunk 1 above)
Comment (scsides) [reply CHANGELOG.md]: Sounds good, I'll remove it.

**Comment by jessemapel** [comment]
For some reason this is failing on OSX with Kakadu because it uses a `.so` extension:

```
ld: library not found for -llibkdu_a79R.so
```

and in the linker command: 

-lc++ -lpthread -lz -ldl -lm **-llibkdu_a79R.so -llibkdu_v79R.so** -lnn -lopencv_calib3d -lopencv_core

**Comment by scsides** [comment]
Yes, I saw that and will work on a fix before changing from draft PR

**Review by jessemapel — APPROVED** [review]
Looks good. The special handling is disappointing, but it looks like it's necessary. Hopefully we can move to target based linking in the future to avoid this stuff.

**Review by krlberry — APPROVED** [review]
:+1: Looks good to me.

**Pull request merged**